### PR TITLE
Unit test improvements

### PR DESCRIPTION
**Pull request by Blinke**
I've got a working implementation of the test infrastructure now. The setup is a bit weird in places due to the reliance on data from the PoE API and the whole initialization flow but it works for now.

Some basic tests are added, mostly from the debug commands as they illustrated some edge cases that needed to be verified. I want to add more detailed tests before this gets merged so any suggestions of special cases that need to be tested would help me out a lot.

### CODE REVIEW

File: src/Sidekick.Business.Tests/ItemParserTests/EquipmentParsing.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                actual.Sockets
+                    .Should().HaveCount(6)
+                    .And.OnlyContain(socket => socket.Group == 0);
+            }
+        }
+
+
+        #region ItemText
+
+        private const string UniqueSixLink = @"Rarity: Unique
```
Comment (stormbringer766) [inline src/Sidekick.Business.Tests/ItemParserTests/EquipmentParsing.cs:41]: @Blinke I think it would be better if these are inline with the test. Uglifies the test but you can see what is being tested.

Comment (Blinke) [reply src/Sidekick.Business.Tests/ItemParserTests/EquipmentParsing.cs:41]: Since they are constants, you can mouse over them in Visual Studio and see the value. I think that's a good balance between cleanliness and readability.

**Comment by domialex** [comment]
I think we might want to support other languages also, that's something that can break easily and would be useful to test.

**Comment by Blinke** [comment]
> I think we might want to support other languages also, that's something that can break easily and would be useful to test.

I agree that it would be good to have in the long run. I believe that's out of the scope of this PR though.

**Comment by leMicin** [comment]
I would move the test projects physically. I receommend these changes:
- /src/Sidekick.Business.Tests to /tests/Sidekick.Business.Tests
- /src/Sidekick.TestDataCollector to /tests/Sidekick.TestDataCollector
- /src/Sidekick.TestInfrastructure to /tests/Sidekick.TestInfrastructure

**Comment by leMicin** [comment]
I made the changes myself.

@Blinke If this is ready for merge, please mark it as ready. :)

**Comment by Blinke** [comment]
I'm going to add some more tests tomorrow and then I'll call it done :)

**Comment by Blinke** [comment]
I added a few more test I could think of, this should be good enough for now.

@domialex @leMicin

File: src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/Pseudo/PseudoStatDataService.cs [hunk 2]
```diff
@@ -38,15 +34,13 @@ public async Task OnInit()
             {
                 logger.Information($"Pseudo stat service initialization started.");
 
-                var response = await client.GetAsync("https://www.pathofexile.com/api/trade/data/stats/");
-                var content = await response.Content.ReadAsStreamAsync();
-                var result = await JsonSerializer.DeserializeAsync<FetchResult<StatDataCategory>>(content, poeApiClient.Options);
+                var result = await poeApiClient.Fetch<StatDataCategory>();
```
Comment (domialex) [inline src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/Pseudo/PseudoStatDataService.cs:37]: What was done here is that we always fetched the English stats and gather the Ids with regexes below in `InitGroups`. With this change here, we actually fetch in other languages so the regexes won't work anymore for any language other than English.
We wanted to avoid having to maintain other languages here (the regexes) since we only really need to do this once and compare ids like we do on `FillPseudo` in `StatDataService`.

Comment (Blinke) [reply src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/Pseudo/PseudoStatDataService.cs:37]: Ahh, yeah I missed that completely. I'll try to figure out a good solution for this

Comment (domialex) [reply src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/Pseudo/PseudoStatDataService.cs:37]: I might be wrong, but you can just revert this part and it should work.

Comment (Blinke) [reply src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/Pseudo/PseudoStatDataService.cs:37]: The reason I changed it was to make use of the cached data from the PoE api and not make new requests each time the tests are run. Should be able to find a workaround though.

Comment (Blinke) [reply src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/Pseudo/PseudoStatDataService.cs]: I finally got around to fixing this. Ended up needing to make some changes to to the language provider to access the default language but I think it's fine for now. The language provider is a bit finicky work with at the moment and could probably do with a bit of cleaning up in the future.

Comment (domialex) [reply src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/Pseudo/PseudoStatDataService.cs]: Cool, just message me when you feel it's ready.

**Comment by Blinke** [comment]
@domialex It's ready for merge now :)
